src/amltk/pipeline/node.py

```diff
@@ -776,3 +865,508 @@ def _fufill_param_requests(
+                Additional plugins to attach to the eventual
+                [`Task`][amltk.scheduling.task.Task] that will be executed by
+                the [`Scheduler`][amltk.scheduling.Scheduler]. Please
+                refer to the
+                [plugins reference](../../../reference/scheduling/plugins.md) for more.
+            setup_only:
+                Whether to only perform setup of the optimization but do
+                not call [`scheduler.run()`][amltk.scheduling.Scheduler.run].
+                This can be useful when you'd like to setup multiple pipelines to
+                optimize or building more complex optimization procedures.
+                When `True`, the `scheduler=` that was passed in or the one
+                created will be returned instead of the history.
+        """  # noqa: E501
+        match history:
+            case None:
+                history = History()
+            case History():
+                pass
+            case _:
+                raise ValueError(f"Invalid history {history}. Must be a History")
+
+        _plugins: tuple[Plugin, ...]
+        match plugins:
+            case None:
+                _plugins = ()
+            case Plugin():
+                _plugins = (plugins,)
+            case Iterable():
+                _plugins = tuple(plugins)
+            case _:
+                raise ValueError(
+                    f"Invalid plugins {plugins}. Must be a Plugin or an Iterable of"
+                    " Plugins",
+                )
+
+        if any(
+            limit is not None
+            for limit in (
+                process_memory_limit,
+                process_walltime_limit,
+                process_cputime_limit,
+            )
+        ):
+            try:
+                from amltk.scheduling.plugins.pynisher import PynisherPlugin
+            except ImportError as e:
+                raise ImportError(
+                    "You must install `pynisher` to use `trial_*_limit`"
+                    " You can do so with `pip install amltk[pynisher]`"
+                    " or `pip install pynisher` directly",
+                ) from e
+            # TODO: I'm hesitant to add even more arguments to the `optimize`
+            # signature, specifically for `mp_context`.
+            plugin = PynisherPlugin(
+                memory_limit=process_memory_limit,
+                walltime_limit=process_walltime_limit,
+                cputime_limit=process_cputime_limit,
+            )
+            plugins = (*_plugins, plugin)
+
+        # If threadpool_limit_ctl None, we should default to inspecting if it's
+        # an sklearn pipeline. This is because sklearn pipelines
+        # run in parallel will over-subscribe the CPU and cause
+        # the system to slow down.
+        # We use a heuristic to check this by checking if the item at the head
+        # of this node is a subclass of sklearn.base.BaseEstimator
+        match threadpool_limit_ctl:
```

Comment:
I swear, you just wrote this function because you wanted to use match cases :D

Reply:
You should see some tool I'm building with AMLTK ;) Most of it is just parsing user arguments and if it doesn't match anything throw an error. Still better than a truck ton of `if`s :)